src/psyclone/tests/extract_test.py

```diff
@@ -0,0 +1,828 @@
+    # try to extract the region between the Directive and the
+    # exterior Loop over colours
+    schedule, _ = otrans.apply(colour_loop)
+    directive = schedule.children[0].children[0]
+    with pytest.raises(TransformationError) as excinfo:
+        _, _ = etrans.apply(directive)
+    assert ("Dynamo0.3 API: Extraction of a Loop over cells in a "
+            "colour without its ancestor Loop over colours is not "
+            "allowed.") in str(excinfo)
+
+
+def test_no_outer_loop_gocean1p0():
+    ''' Test that applying ExtractRegionTrans on an inner Loop without its
+    parent outer Loop in GOcean1p0 API raises a TransformationError. '''
+    etrans = ExtractRegionTrans()
+
+    _, invoke_info = parse(os.path.join(GOCEAN_BASE_PATH,
+                                        "single_invoke_three_kernels.f90"),
+                           api=GOCEAN_API)
+    psy = PSyFactory(GOCEAN_API, distributed_memory=False).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    schedule = invoke.schedule
+
+    # Try to extract the region between the outer and the inner Loop
+    inner_loop = schedule.children[0].children[0]
+    with pytest.raises(TransformationError) as excinfo:
+        _, _ = etrans.apply(inner_loop)
+    assert ("GOcean1.0 API: Extraction of an inner Loop without its "
+            "ancestor outer Loop is not allowed.") in str(excinfo)
+
+# --------------------------------------------------------------------------- #
+# ================== ExtractNode tests ====================================== #
+# --------------------------------------------------------------------------- #
+
+
+def test_extract_node_position():
+    ''' Test that ExtractRegionTrans inserts the ExtractNode in the position
+    of the first Node a Schedule in the Node list marked for extraction. '''
+    etrans = ExtractRegionTrans()
+
+    # Test GOcean1.0 API for extraction of a single Node
+    _, invoke_info = parse(os.path.join(GOCEAN_BASE_PATH,
+                                        "single_invoke_three_kernels.f90"),
+                           api=GOCEAN_API)
+    psy = PSyFactory(GOCEAN_API, distributed_memory=False).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    schedule = invoke.schedule
+    # Apply Extract transformation to the second Node and assert that
+    # position and the absolute position of the ExtractNode are the same as
+    # respective positions of the second Node before the transformation.
+    pos = 1
+    child = schedule.children[pos]
+    abspos = child.abs_position
+    dpth = child.depth
+    schedule, _ = etrans.apply(child)
+    extract_node = schedule.walk(schedule.children, ExtractNode)
+    # The result is only one ExtractNode in the list with position 1
+    assert extract_node[0].position == pos
+    assert extract_node[0].abs_position == abspos
+    assert extract_node[0].depth == dpth
+
+    # Test Dynamo0.3 API for extraction of a list of Nodes
```

Comment:
I am not testing the APIs here, I am testing extraction of 1) a single node using and example from GOcean 1.0 API and 2) a list of nodes using an example from Dynamo 0.3 API. It could have been the opposite.

Reply:
Oh I see. Great.